src/community/wps-download/src/main/java/org/geoserver/wps/gs/download/GridGeometryProvider.java

```diff
@@ -505,10 +513,8 @@ private GridGeometry2D computeGridGeometry2D(
         // requested gridGeometry such that the resulting scale doesn't perfectly match the
         // requested resolution
         if (Math.abs(scaleX - resolution[0]) > 1E-6 || (Math.abs(scaleY - resolution[1]) > 1E-6)) {
-            if (crsRequestHandler != null
-                    && crsRequestHandler.getReferenceFeatureForAlignment() != null) {
-                SimpleFeature referenceFeature =
-                        crsRequestHandler.getReferenceFeatureForAlignment();
+            SimpleFeature referenceFeature = crsRequestHandler.getReferenceFeatureForAlignment();
```

Comment:
crsRequestHandler is not checked for null anymore, is it guaranteed to be non null here?

Reply:
> > (1): the WPS main zip package must not include wps-download, it should be in its own separate zip (not sure about hazelcast, it also has its own zip, should not be there either....)
> > ```
> 
> Note that they are actually on an flag indeed.
> I had to put it gs-wps-download into an exclude section because I tested the release zip and found it within the wps-plugin.zip.
> So I have used that which is actually what are you suggesting, right?

Ah yes, did not see they were in exclude. Good.